src/redis.js

```diff
@@ -0,0 +1,61 @@
+var redis = require('redis');
+var debug = require('debug')('feathers-sync:redis');
+
+var pub = redis.createClient();
+var sub = redis.createClient();
+
+module.exports = function(config) {
+  debug('setting up database %s', config.db);
+
+  return function() {
+
+    var oldSetup = this.setup;
+
+    this.setup = function() {
+
+      var result = oldSetup.apply(this, arguments);
+      var services = this.services;
+
+      Object.keys(services).forEach(function(path) {
+
+        var service = services[path];
+        service._serviceEvents.forEach(function(event) {
+
+          var ev = path + ' ' + event;
+          debug('subscribing to handler %s', ev);
+
+          sub.subscribe(ev);
+          sub.on('message', function(data) {
+            if(data !== ev ){ return; }
+            debug('got event, calling old emit %s', data);
+            service._emit.call(service, event, data);
+          });
```

Comment:
This is not ideal. We're binding for each path and event, then filtering the events based on scope but redis doesn't give us one listener per channel, all channels share the same `message` event. This means we're adding exponentially more listeners then we need to here then filtering out the ones we don't care about. 

I want purge this entire file and just use [socket.io-redis](http://github.com/socket.io/socket.io-redis) however I don't see to have access to `io.adapter` here. I've done it previously but I had to mount the redis adapter inside the `app.configure(socketio(adapter))` rather then inside a `feathers-sync` specific module.